### PR TITLE
[1LP][RFR] Fix for GCE provider

### PR DESCRIPTION
**Pull request by vatsalparekh**
Purpose or Intent
=================
`security_group` is not found in GCE provider, adding conditions for handling `environment` dict 
{{ pytest: -vvvv cfme/tests/services/test_cloud_service_catalogs.py --use-provider complete }}

### CODE REVIEW

**Comment by vatsalparekh** [comment]
@mshriver review please

**Comment by dajoRH** [comment]
Lint report for commit eee15aca5ce7d2b9ffb3fb43d3b2f48eb19d0720:

`cfme/tests/services/test_cloud_service_catalogs.py`:
- :bangbang: Line 50:49: **E127** *continuation line over-indented for visual indent*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by lkhomenk** [comment]
added `--use-provider complete` for PRT run

File: cfme/tests/services/test_cloud_service_catalogs.py [hunk 1]
```diff
@@ -41,11 +43,14 @@ def test_cloud_catalog_item(appliance, setup_provider, provider, dialog, catalog
...
         'properties': {'instance_type': partial_match(provisioning['instance_type']),
                        },
-        'environment': {'security_groups': partial_match(provisioning['security_group']),
-                        }
+        'environment': {}
     }
+    if not provider.one_of(GCEProvider):
+        provisioning_data['environment'] = {'security_groups':
+                                            partial_match(provisioning['security_group']),
+                                            'cloud_tenant': provisioning['cloud_tenant']}
```
Comment (lkhomenk) [inline cfme/tests/services/test_cloud_service_catalogs.py]: REQUIRED: Azure doesn't have 'cloud_tenant'

**Comment by vatsalparekh** [comment]
The errors are not random, and mainly when provider was added but options not found yet in the UI.
This PR solves issue of adding conditions for providers and elements that don't apply to them in the UI form.

**Comment by psav** [comment]
So what are the failures? I'm a little confused

**Comment by vatsalparekh** [comment]
@psav the failures seems like that provider was added but the details in the form seems empty, maybe provider didn't fetch them yet. The changes here necessary because the details such as `security_group` or `cloud_tenant` is not applicable to GCE and Azure.

**Pull request merged**